packages/orbit-components/config/fetchIllustrations.mts

```diff
@@ -0,0 +1,198 @@
+import ora from "ora";
+import dedent from "dedent";
+import { path, fs, chalk, $, fetch, argv } from "zx";
+import _ from "lodash";
+import filedirname from "filedirname";
+
+import { dotenv, api } from "./helpers.mjs";
+import type { FigmaComponents, Component } from "./figma.d";
+import { NAMES as CURRENT_ILLUSTRATION_NAMES } from "../src/Illustration/consts.mjs";
+import { generateTypeFile } from "./typeFiles.mjs";
+
+dotenv();
+
+const gitlabRestApi = (url: string, method: "POST" | "GET", body?: string) => {
+  return fetch(url, {
+    method,
+    headers: {
+      "Content-Type": "application/json",
+      Authorization: `Bearer ${process.env.GITLAB_TOKEN}`,
+    },
+    body,
+  }).then(res => {
+    if (!res.ok) {
+      throw new Error(`Gitlab API error: ${res.status} ${res.statusText}`);
+    }
+
+    return res.json();
+  });
+};
+
+const PROJECT_ID = encodeURIComponent("frontend/images.kiwi.com");
+const GITLAB_API_MERGE_REQUEST = `https://gitlab.skypicker.com/api/v4/projects/${PROJECT_ID}/merge_requests`;
+const GITLAB_API_COMMIT = `https://gitlab.skypicker.com/api/v4/projects/${PROJECT_ID}/repository/commits`;
+const GITLAB_API_BRANCH = `https://gitlab.skypicker.com/api/v4/projects/${PROJECT_ID}/repository/branches`;
+const BRANCH_NAME = "feat/orbit-illustrations";
+
+const ILLUSTRATION_NAMES = [
```

Comment:
> First, this can remove names from the types just because images are removed from Figma.

It will create PR and allow it or not - on us, so I would not be afraid about that.

> The AirportTransportBus name is added but that illustration does not exist (yet?) on images.kiwi

the part of this script is also to upload it there, create MR and approve it. The process is not aligned 100%, it might happen that we release it in Orbit, but it's still not approved on images.kiwi, but it's not a big problem, more like a process solving in my opinion, it's two clicks to merge MR on images.kiwi and we always can make sure to release after. 

> Second, it is an automation that only does part of the job, which is the less costly one: adding a name. Whenever there is a new illustration there are more steps to be done that this does not cover (at least for now). Besides that, it is unable to detect updates to images, which do not require any work in this repo, but require work on the images.kiwi repo

it's not, it also cares about image upload. I would not really do it for now so complicated (I mean checking if the image was updated, not the name of the image or added the new one), for the beginning having an upload and type updated is a good start to automate it. 

> The CovidTest illustration exists in images.kiwi. Further investigation led me to it being used on the mailing project, with the illustration url directly.

no problem, it's not increasing the bundle size on our side and it's a part of our Orbit library, so should be added

Reply:
> Either way, I feel this process is not so complicated and it seems a bit of overengineering to be automating this, especially because it's just a partial automation

it's not overengineering, it's already automated in the android team and works good for them. For as it can help to reduce the time with adding new images and the need to communicate with designers, as you mentioning it above, sometimes images are not uploaded to Orbit react, even when they are part of our library. Although It might be better solved via uploading it directly to AWS.

Reply:
I would improve it, so it won't create for us any troubles, but the benefits are even having it as it is proposed now.

Reply:
> 

it's the images that are not present inside Figma, but on images.kiwi

Reply:
> What are these names?

it's the images that are not present inside Figma, but on images.kiwi